example_test.go

```diff
@@ -0,0 +1,32 @@
+package xsync_test
+
+import (
+	"encoding/binary"
+	"hash/maphash"
+	"time"
+
+	"github.com/puzpuzpuz/xsync"
+)
+
+func ExampleNewTypedMapOf() {
+	type Person struct {
+		GivenName   string
+		FamilyName  string
+		YearOfBirth int
+	}
+	age := xsync.NewTypedMapOf[Person, int](func(seed maphash.Seed, p Person) uint64 {
+		var h maphash.Hash
+		h.SetSeed(seed)
+		// We write field numbers, to make sure that "foo", "bar" and "foob", "ar" hash differently.
```

Comment:
Field numbers won't protect from hash collision attacks. After all, bytes 0, 1 and so on are valid ASCII chars. The latest version of this example contains a proper implementation of a hash function for a struct.

Decided to leave this comment for those who would be looking at this PR.

Reply:
True. Should've done something with length-prefixing or the like. Oh well.